packages/focusjail/src/useFocusJail.ts

```diff
@@ -6,45 +6,28 @@
  */
 
 import { createRef, useRef, useEffect, useState, useCallback } from 'react';
-import { composeEventHandlers, KEY_CODES } from '@zendeskgarden/container-utilities';
+import { composeEventHandlers, KEYS } from '@zendeskgarden/container-utilities';
 import { tabbable } from 'tabbable';
 import activeElement from 'dom-helpers/activeElement';
+import { IUseFocusJailProps, IUseFocusJailReturnValue } from './types';
 
-export interface IUseFocusJailProps {
-  /** Focuses on the `containerRef` element after mounting */
-  focusOnMount?: boolean;
-  /** Determines whether to return keyboard focus to the last active element upon unmounting */
-  restoreFocus?: boolean;
-  /** The global environment where the focus jail is rendered */
-  environment?: Document;
-  /** A callback function that receives the focused element */
-  focusElem?: (element: HTMLElement) => any;
-  /** A [ref](https://reactjs.org/docs/refs-and-the-dom.html) pointing to the focus jail's container element */
-  containerRef: React.RefObject<HTMLElement>;
-}
-
-export interface IUseFocusJailReturnValue {
-  getContainerProps: <T>(options?: T) => T & React.HTMLProps<any>;
-  focusElement: (element?: HTMLElement) => any;
-}
-
-export const useFocusJail = (
+export const useFocusJail = <T extends Element = Element>(
   {
     focusOnMount = true,
     restoreFocus = true,
-    environment,
+    environment = document,
```

Comment:
Are other uses of `environment` throughout the codebase problematic? This pattern is repeated.

Reply:
Also are you sure a `ReferenceError` is thrown here? No dereferencing is happening and the `document` object is simply `undefined` in SSR. I think the default is only evaluated when it is not provided (which it would need to be in a SSR environment).

Reply:
Either way, replaced with code that is less intrusive to the original structure and safely tucks `document` within a `useEffect`.

Reply:
This is probably out of scope of this PR.. apologies to throw a wrench at things here... I think the team should come together to discuss this and come up with a strategy for this in the future.

> Also are you sure a ReferenceError is thrown here? No dereferencing is happening and the document object is simply undefined in SSR. I think the default is only evaluated when it is not provided (which it would need to be in a SSR environment).

No I'm not certain since I did not test this in a SSR environment.. however if the `useEffect` is ran (which IIRC shouldn't), it would be using `document.body` ([here](https://github.com/zendeskgarden/react-containers/pull/461/files/dd407bb20264a377647e69647ad6f8506a6d9f89#diff-d68d3e3b7d94b05e530309f020968d36263ec9e51e65595b0bbfb52bb8a40894R114)) and `document.activeElement` ([here](https://github.com/zendeskgarden/react-containers/pull/461/files/dd407bb20264a377647e69647ad6f8506a6d9f89#diff-d68d3e3b7d94b05e530309f020968d36263ec9e51e65595b0bbfb52bb8a40894R107)). I think this threw me off initially.